### PR TITLE
Cache surprisingly slow findVariables calls

### PR DESCRIPTION
**Pull request by darabos**
Other functions in this file that run code are already cached, but I guess this was not slow enough. But I now have a workspace where it would take a couple seconds to just open the parameters of a Derive box. This fixed it.

### CODE REVIEW

File: app/com/lynxanalytics/sandbox/ScalaScript.scala [hunk 1]
```diff
@@ -286,17 +286,22 @@ object ScalaScript {
     }
   }
 
+  private val findVariablesCache = new SoftHashMap[String, Set[String]]()
```
Comment (olahg) [inline app/com/lynxanalytics/sandbox/ScalaScript.scala:289]: Is this thread-safe?

Comment (olahg) [reply app/com/lynxanalytics/sandbox/ScalaScript.scala:289]: Sorry for the stupid question. It is thread-safe.

**Review by olahg — APPROVED** [review]
LGTM

File: app/com/lynxanalytics/sandbox/ScalaScript.scala (hunk 1 above)
Comment (darabos) [reply app/com/lynxanalytics/sandbox/ScalaScript.scala:289]: > Sorry for the stupid question. It is thread-safe.

Oh, thanks for checking! I just copied it from the other cached things in this file.

**Pull request merged**